### DISCUSSION TITLE
Differences in Model Behavior for 8kHz and 16kHz Audio Inputs

### DISCUSSION THREAD

**Discussion by tamarabanovac** [snakers4/silero-vad#575] [Q&A] [answered]
Hello,

I know that Silero VAD supports both 8kHz and 16kHz audio data. I am curious about the differences in model behavior when processing these two sampling rates. Specifically, I would like to understand:

1. Are there any concrete differences in how the model operates when given audio data at these two sampling rates, or is there just an upsampling to 16kHz happening internally?
2. Was the model trained on both 16kHz and 8kHz audio data, or is it specifically optimized for one of the rates (such as 16kHz)?

Thank you for any insights or clarifications!

**Comment by snakers4** [comment, marked as answer]
Hi,

> Are there any concrete differences in how the model operates when given audio data at these two sampling rates, or is there just an upsampling to 16kHz happening internally?

For previous model versions there was some difference in quality.
https://github.com/snakers4/silero-vad/issues/2#issuecomment-2195433115

There was a chart on old wiki pages, but it's gone now.
For the latest model it is kind of the same.
If I am not mistaken, both the JIT and the ONNX models contain 2 actual models - one for 8k and one for 16k.
Hence the problem with ONNX export with opset < 16.

> Was the model trained on both 16kHz and 8kHz audio data, or is it specifically optimized for one of the rates (such as 16kHz)?

The model was trained by resampling into either 16k or 8k.

Also internally higher sample rates like 32k or 48k that are a multiple of 16k are naively resampled - i.e. each n-th sample is taken.

**Reply by tamarabanovac** [reply]
Thanks for the explanation!
